### ISSUE TITLE
Recovered from panic

### ISSUE THREAD

**Issue by mattkbach** [irvinlim/apple-health-ingester#20]
Hello

Just starting getting this error:

```
applehealth        | 2023-01-31T18:08:44.537079798Z time="2023-01-31T18:08:44Z" level=error msg="recovered from panic in backend:\ngoroutine 8 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/irvinlim/apple-health-ingester/pkg/ingester.(*Ingester).processWriteItem.func1()\n\t/go/src/workspace/pkg/ingester/ingester.go:168 +0x197\npanic({0x7541c0, 0xa827b0})\n\t/usr/local/go/src/runtime/panic.go:1038 +0x215\ngithub.com/irvinlim/apple-health-ingester/pkg/backends/influxdb.makeSleepPoint({0x7d9521, 0x17}, {0xc000074f30, 0x29}, {0x0, 0x0}, 0x1, 0x0, 0x0, {0xc0001c3388, ...})\n\t/go/src/workspace/pkg/backends/influxdb/backend.go:171 +0x160\ngithub.com/irvinlim/apple-health-ingester/pkg/backends/influxdb.(*Backend).getMetricPoints(0xc000190770, 0xc0002bf420, {0xc0001c3388, 0x1, 0x1})\n\t/go/src/workspace/pkg/backends/influxdb/backend.go:146 +0x409\ngithub.com/irvinlim/apple-health-ingester/pkg/backends/influxdb.(*Backend).writeMetrics(0xc000020b40, {0xc000114d80, 0x60, 0x8d}, {0x0, 0x0})\n\t/go/src/workspace/pkg/backends/influxdb/backend.go:99 +0x3eb\ngithub.com/irvinlim/apple-health-ingester/pkg/backends/influxdb.(*Backend).Write(0x683a0b, 0xc00000e260, {0x0, 0x0})\n\t/go/src/workspace/pkg/backends/influxdb/backend.go:70 +0x72\ngithub.com/irvinlim/apple-health-ingester/pkg/ingester.(*Ingester).processWriteItem(0xc000028500, {0x7296c0, 0xc0002f1c20}, {0x84e918, 0xc000028520})\n\t/go/src/workspace/pkg/ingester/ingester.go:173 +0xd3\ngithub.com/irvinlim/apple-health-ingester/pkg/ingester.(*Ingester).processQueue(0xc00014c480, 0xc000028520)\n\t/go/src/workspace/pkg/ingester/ingester.go:138 +0x16a\ncreated by github.com/irvinlim/apple-health-ingester/pkg/ingester.(*Ingester).Start\n\t/go/src/workspace/pkg/ingester/ingester.go:59 +0x71\n" backend=InfluxDB payload="&{0xc00000e260 }"
applehealth        | 2023-01-31T18:08:44.537662340Z time="2023-01-31T18:08:44Z" level=error msg="write data error" backend=InfluxDB elapsed=113.132516ms error="recovered from panic"
```

Not sure but seems to maybe have something to do with the latest update of the Auto Export app and the changes in sleep data that is exported? It was working fine yesterday and I woke up this morning to a new app update and these errors in the logs.

Also just wanted to take a moment and say thank you for developing and maintaining this! Very much appreciated :)

**Comment by irvinlim** [comment]
Hello! I have indeed confirmed that this is an issue when used with Health Auto Export 6.6.2 onwards, and when `Aggregate Sleep Analysis` is turned on. If the toggle is turned off, the panic won't be triggered.

There are several non-backwards compatible changes in the data structures exposed by HAE, which results in a few things:

1. Triggers a panic bug due to non-robust handling of JSON fields. Aggregated sleep analysis datapoints are wrongly recognized as non-aggregated sleep analysis datapoints here (due to the addition of the `source` field): https://github.com/irvinlim/apple-health-ingester/blob/a3877bafe8c0119fdee2ef1968feba038b5e52ef/pkg/healthautoexport/types.go#L114-L133
2. Removal of `inBedStart`/`inBedEnd` fields from aggregated sleep analysis datapoints.

However, we now have access to sleep phase data from the recent additions in Apple Health: https://crescent.co/blog/apple-watch-sleep-stages-watchos9

I've started work on a fix, so do keep a lookout for it!
